janusgraph-core/src/main/java/org/janusgraph/graphdb/database/StandardJanusGraph.java

```diff
@@ -230,6 +248,37 @@ public String getGraphName() {
         return this.name;
     }
 
+    @Override
+    public Object eval(String gremlinScript, boolean commit) {
```

Comment:
I think I would better rename `boolean commit` to `boolean readOnlyBoundTraversal` (or some similar name) so that it would be clear that the bound traversal is created with read-only mode.
In Javadoc or JanusGraph documentation I would also add information that this method commits a newly created transaction when `readOnlyBoundTraversal == false` or rollbacks a newly created transaction when `readOnlyBoundTraversal == true`.
The reason for that is because we will be able to extend implementations later by adding a new method like:
`eval(String script, boolean readOnlyBoundTraversal, boolean wrapTraversalInNewTx)`

Then an implementation of previous `eval` will look like:
```
public Object eval(String gremlinScript, boolean readOnlyBoundTraversal) {
  return eval(gremlinScript, readOnlyBoundTraversal, true);
}
```

Reply:
`readOnlyBoundTraversal` is ambiguous IMO coz non-readonly does not necessarily mean it will be committed.

I do see your point here, that's a valid concern. 👍 My feeling is, once we allow users to use existing threaded transaction, it would be insufficient to use a combination of binary flags to control. My proposal is, once we extend implementation later, we could use an enum to control:

- `newTxReadOnly`: open a new transaction and is read-only
- `newTxAutoCommit`: open a new transaction and do auto-commit
- `threadedTxReadOnly`: use existing threaded tx (could see uncommitted results) and is read-only
- `threadedTxAutoCommit`: use existing threaded tx (could see uncommitted results) and do auto-commit
- `threadedTxNormal`: use existing threaded tx (could see uncommitted results), read is allowed, but no auto-commit

As of now, I am inclined to using the name `commit` rather than sth like `readOnly`. I will add documentation once this PR is merged for sure.

Reply:
I guess instead of developing different variants of `eval` method and try to guess all possible use-cases we could teach people in the documentation on how to create their own script evaluation engine which isn't restricted to a specific use-case.
I.e. I would imagine a simple documentation which points to TinkerPop documentation and then maybe adds some examples like:
```
JanusGraph graph = ...;
GremlinLangScriptEngine scriptEngine = new GremlinLangScriptEngine();
Bindings bindings = scriptEngine.createBindings();
bindings.put("graph", graph);
bindings.put("g", graph.traversal());
```

Reply:
> I guess instead of developing different variants of eval method and try to guess all possible use-cases we could teach people in the documentation on how to create their own script evaluation engine which isn't restricted to a specific use-case.

That's a good point! I'd still like to have this method within JG so that it's handy to use. My own usecase only involves read-only transactions so I am not even sure if we shall expose the auto-commit functionality or not. What do you think?

Reply:
> My own usecase only involves read-only transactions so I am not even sure if we shall expose the auto-commit functionality or not. What do you think?

To be fair I don't have much preference on that. If we add good documentation I'm OK with any case (i.e. with `commit` and without `commit`).